src/__tests__/app/create-email-template/page.test.tsx

```diff
@@ -1,9 +1,69 @@
 import CreateEmailTemplatePage from '@app/create-email-template/[sessionId]/page';
-import { render } from '@testing-library/react';
+import { getSession } from '@utils/form-actions';
+import { Session, TemplateType } from '@utils/types';
+import { redirect } from 'next/navigation';
+import { CreateEmailTemplate } from '@forms/CreateEmailTemplate/CreateEmailTemplate';
 
-test('CreateEmailTemplatePage', async () => {
-  const page = await CreateEmailTemplatePage();
-  const container = render(page);
+jest.mock('@utils/amplify-utils', () => ({
+  getAmplifyBackendClient: () => ({
+    models: {
+      SessionStorage: {
+        update: () => ({ data: {} }),
+      },
+    },
+  }),
+}));
+jest.mock('@utils/form-actions');
+jest.mock('next/navigation');
+jest.mock('@forms/CreateEmailTemplate/CreateEmailTemplate');
 
-  expect(container.asFragment()).toMatchSnapshot();
+const getSessionMock = jest.mocked(getSession);
+const redirectMock = jest.mocked(redirect);
+const CreateEmailTemplateMock = jest.mocked(CreateEmailTemplate);
+
+const initialState: Session = {
+  id: 'session-id',
+  templateType: TemplateType.EMAIL,
+  nhsAppTemplateName: '',
+  nhsAppTemplateMessage: '',
+};
+
+describe('CreateEmailTemplatePage', () => {
+  beforeEach(jest.resetAllMocks);
+
+  it('should redirect to invalid-session when no session is found', async () => {
+    getSessionMock.mockResolvedValueOnce(undefined);
+
+    await CreateEmailTemplatePage({ params: { sessionId: 'session-id' } });
+
+    expect(getSessionMock).toHaveBeenCalledWith('session-id');
+
+    expect(redirectMock).toHaveBeenCalledWith('/invalid-session', 'replace');
+  });
+
+  it('should redirect to invalid-session when sessions template type is not EMAIL', async () => {
+    getSessionMock.mockResolvedValueOnce({
+      ...initialState,
+      templateType: TemplateType.NHS_APP,
+    });
+
+    await CreateEmailTemplatePage({ params: { sessionId: 'session-id' } });
+
+    expect(getSessionMock).toHaveBeenCalledWith('session-id');
+
+    expect(redirectMock).toHaveBeenCalledWith('/invalid-session', 'replace');
+  });
+
+  it('should render CreateEmailTemplatePage component when session is found', async () => {
+    getSessionMock.mockResolvedValueOnce(initialState);
+    CreateEmailTemplateMock.mockImplementationOnce(() => <p>rendered</p>);
+
+    const page = await CreateEmailTemplatePage({
+      params: { sessionId: 'session-id' },
+    });
+
+    expect(getSessionMock).toHaveBeenCalledWith('session-id');
+
+    expect(page).toMatchSnapshot();
```

Comment:
You should wrap the page in a `render(...)` function call.

Reply:
Added to https://nhsd-jira.digital.nhs.uk/browse/CCM-6950 noticed this is the case in multiple pages already.